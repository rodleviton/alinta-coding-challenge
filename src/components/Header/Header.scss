.header {
  position: fixed;
  background: #fff;
  border-top: 2px solid #EF7F11;
  border-bottom: 1px solid #eee;
  height: 70px;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 15;
}

.logo {
  background: url('./img/logo.png') no-repeat;
  background-size: 160px 39px;
  overflow: hidden;
  margin: 13px 0 0 20px;
  width: 160px;
  height: 42px;
}

@media (max-width: 600px) {
  .logo {
    width: 42px;
  }
}

// Github icon
.github {
  position: fixed;
  z-index: 15;
  right: 2.5rem;
  top: 2rem;
  width: 33px;
  height: 32px;
  background: url('./img/github.svg') no-repeat;
  background-size: 33px 32px;
  display: inline-block;
}

.avatar {
  position: absolute;
  z-index: 5;
  top: 32px;
  left: 50%;
  margin-left: -32px;
  width: 64px;
  height: 64px;
  border-radius: 50%;
  border: 1px solid #d2d2d2;
  display: block;
  background: url('./img/rodleviton.png') no-repeat;
  background-size: 64px 64px;

  &:hover, &:focus {
    text-decoration: none;
  }
}
