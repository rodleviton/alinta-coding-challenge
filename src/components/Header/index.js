import React from 'react';
import styles from './Header.scss';

/**
 * Component to display main page header
 */
const Header = () => {
  return (
    <header className={styles.header}>
      <div className={styles.logo}></div>

      <div className={styles.avatar}></div>

      <a href="https://github.com/rodleviton/alinta-code-challenge" title="Rod Leviton Github Repo" target="_blank" className={styles.github}></a>
    </header>
  );
};

export default Header;
