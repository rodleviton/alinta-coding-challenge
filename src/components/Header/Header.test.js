import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Header from './';

describe('Header Component', () => {
  it('should render correctly', () => {
    const tree = ReactTestRenderer.create(
      <Header />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
