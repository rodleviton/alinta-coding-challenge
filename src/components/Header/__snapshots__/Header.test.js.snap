// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<header
  className={undefined}
>
  <div
    className={undefined}
  />
  <div
    className={undefined}
  />
  <a
    className={undefined}
    href="https://github.com/rodleviton/alinta-code-challenge"
    target="_blank"
    title="Rod Leviton Github Repo"
  />
</header>
`;
